cmake/CompilerFlags.cmake

```diff
@@ -104,6 +104,7 @@ ELSEIF ( CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"
     ADD_CXX_DEFINITIONS("-pipe") # Faster compiler processing
     ADD_CXX_DEFINITIONS("-pedantic") # Turn on warnings about constructs/situations that may be non-portable or outside of the standard
     ADD_CXX_DEFINITIONS("-Wall -Wextra") # Turn on warnings
+    ADD_CXX_DEFINITIONS("-Werror") # Treat warnings as errors
```

Comment:
I'm curious to see if clang will throw or not. We can remove this line if it's problematic, but it worked on GCC and we might as well turn it on to avoid having warnings sneak into the codebase.

Reply:
@Myoldmopar

Reply:
(Enabling /W4 or /Wall on windows would be interesting too)